crates/storage/provider/src/providers/mod.rs

```diff
@@ -39,8 +39,13 @@ pub struct ShareableDatabase<DB> {
 
 impl<DB> ShareableDatabase<DB> {
     /// create new database provider
-    pub fn new(db: DB, chain_spec: ChainSpec) -> Self {
-        Self { db, chain_spec: Arc::new(chain_spec) }
+    pub fn new(db: DB, chain_spec: Arc<ChainSpec>) -> Self {
+        Self { db, chain_spec }
+    }
+
+    /// Take database back, useful if database is a object.
```

Comment:
Can you clarify here? I may be missing some context, I am not sure what object refers to here

Reply:
Idea was to send DB to `SharableDatabase` and take it back after we are finished. But I made Database implemented on `&` so this is not needed. Will remove it.